x-pack/legacy/plugins/apm/common/runtime_types/transaction_sample_rate_rt/index.ts

```diff
@@ -10,8 +10,8 @@ export const transactionSampleRateRt = new t.Type<number, number, unknown>(
   'TransactionSampleRate',
   t.number.is,
   (input, context) => {
-    const value = Number(input);
-    return value >= 0 && value <= 1 && Number(value.toFixed(3)) === value
+    const value = parseFloat(input as string);
```

Comment:
just curious, why `parseFloat`? because it's more permissive than `Number`?

Reply:
The opposite actually. Due to coercion `Number('') => 0` whereas `parseFloat('') => NaN`.
I wanted to disallow empty values.